src/urllib3/connection.py

```diff
@@ -369,6 +377,58 @@ def request_chunked(
         """
         self.request(method, url, body=body, headers=headers, chunked=True)
 
+    def getresponse(  # type: ignore[override]
+        self,
+        request_url: str,
+        request_method: str,
+        pool: "HTTPConnectionPool",
+        retries: Optional["Retry"],
+        preload_content: bool,
+        decode_content: bool,
+        response_conn: Optional["HTTPConnection"],
+        enforce_content_length: bool,
+    ) -> "HTTPResponse":
```

Comment:
Interesting discovery on `conn` and `response_conn`, that seems like a bug or at least an oversight? Can we only pass one to `getresponse`?

Reply:
@sethmlarson We are only passing one to `getresponse`. Do you mean to `_make_request`? or do you mean to not pass `response_con` to `getresponse`?

To be honest I'm not sure, here is the original code:
https://github.com/urllib3/urllib3/blob/43c372f6eb9f9f94848c7b7645ad19ebf6c047c5/src/urllib3/connectionpool.py#L726-L753

I cant figure out how to handle `reponse_conn` with this refactor. I will have to double-check locally, but I'm pretty sure when I didn't pass `response_conn` to the urllib3 `HTTPResponse` constructor, then stuff broke.

Reply:
Yeah I checked locally @sethmlarson and this breaks stuff, especially around releasing connections. 

```
FAILED test/with_dummyserver/test_connectionpool.py::TestConnectionPool::test_for_double_release - assert 5 == (5 - 1)
```

I feel like this would take a separate effort move the connection releasing logic inside the `HTTPConnection`.